src/detection_win.cpp

```diff
@@ -203,49 +214,21 @@ void EIO_Find(uv_work_t* req) {
 /**********************************
  * Local Functions
  **********************************/
-void ToUpper(char * buf) {
-	char* c = buf;
-	while (*c != '\0') {
-		*c = toupper((unsigned char)*c);
-		c++;
-	}
-}
-
-
-void extractVidPid(char * buf, ListResultItem_t * item) {
-	if(buf == NULL) {
-		return;
-	}
-
-	ToUpper(buf);
-
-	char* string;
-	char* temp;
-	char* pidStr, *vidStr;
-	int vid = 0;
-	int pid = 0;
 
-	string = new char[strlen(buf) + 1];
-	memcpy(string, buf, strlen(buf) + 1);
-
-	vidStr = strstr(string, VID_TAG);
-	pidStr = strstr(string, PID_TAG);
-
-	if(vidStr != NULL) {
-		temp = (char*) (vidStr + strlen(VID_TAG));
-		temp[4] = '\0';
-		vid = strtol (temp, NULL, 16);
-	}
+void ExtractDetails(TCHAR* devicePath, ListResultItem_t* item) {
```

Comment:
Wdyt of adding some tests for this `ExtractDetails` method? I lean towards using this Mocha-like library I made [Macchiato](https://github.com/MadLittleMods/macchiato) but feel free to suggest something better and standard.

I want to make sure we cover all the "Device instance ID/path" possibilities.

Reply:
I believe that's a good idea. I will look into Macchiato.